src/modules/src/Eryph.Modules.VmHostAgent/Networks/HostNetworkCommands.cs

```diff
@@ -35,18 +37,51 @@ public class HostNetworkCommands<RT> : IHostNetworkCommands<RT>
 
             }).ToAff()).TraverseParallel(l=>l)
             select switches;
-
     }
 
+    public Aff<RT, Option<VMSystemSwitchExtension>> GetInstalledSwitchExtension() =>
+        from psEngine in default(RT).Powershell
+        let command = PsCommandBuilder.Create().AddCommand("Get-VMSystemSwitchExtension")
+        from results in psEngine.GetObjectsAsync<VMSystemSwitchExtension>(command).ToAff()
+        select results.Select(r => r.Value)
+            .Where(e => string.Equals(e.Name, EryphConstants.SwitchExtensionName, StringComparison.OrdinalIgnoreCase))
+            .HeadOrNone();
+
     public Aff<RT, Seq<VMSwitchExtension>> GetSwitchExtensions() =>
-        from psEngine in default(RT).Powershell.ToAff()
+        from psEngine in default(RT).Powershell
         from vmSwitchExtensions in psEngine.GetObjectsAsync<VMSwitchExtension>(
             PsCommandBuilder.Create()
                 .AddCommand("Get-VMSwitch")
                 .AddCommand("Get-VMSwitchExtension")
-                .AddParameter("Name", "dbosoft Open vSwitch Extension")).ToAff()
+                .AddParameter("Name", EryphConstants.SwitchExtensionName)).ToAff()
         select vmSwitchExtensions.Map(s => s.ToValue());
 
+    public Aff<RT, Unit> DisableSwitchExtension() =>
+        from psEngine in default(RT).Powershell
+        from _ in psEngine.RunAsync(PsCommandBuilder.Create()
+            .AddCommand("Get-VMSwitch")
+            .AddCommand("Get-VMSwitchExtension")
+            .AddParameter("Name", EryphConstants.SwitchExtensionName)
+            .AddCommand("Disable-VMSwitchExtension")).ToAff()
```

Comment:
Disable-VMSwitchExtension should change switches where the exension is enabled.  The command itself is not able to do the filtering on its own...
In PS:

Get-VMSwitch | Get-VMSwitchExtension -Name "dbosoft Open vSwitch Extension" | where Enabled -eq $true | Disable-VMSwitchExtension

Reply:
I tweaked the logic and also considered that the switch name might not be sufficient to identify the overlay switch. Now, the driver update reenables the switch extension on exactly the switch on which it was disabled.